### ISSUE TITLE
Allow downloading directories

### ISSUE THREAD

**Issue by GrayShade** [aperezdc/ngx-fancyindex#34]
It would be nice to be able to download a complete subdirectory from an index page. The `tar` format would should this without reading the files in memory.

**Comment by aperezdc** [comment]
The potential problem I see with this is that requesting a whole directory may end up in huge transfers being sent to the clients. Not to mention that it would add quite a bit of complexity to the module. I will think about it a bit more, but so far what would be best in my opinion is to put this fuctionality in a separate module.

**Comment by GrayShade** [comment]
I suppose that directory downloads could be disabled by configuration.

I agree about the complexity argument. On the other hand, I'm not familiar to the NGINX module API, but `ngx-fancyindex` seems already complex as-is. Perhaps some of this is inherent to the API? :)

A while ago I looked into implementing something like this for Lighttpd. It seemed to me that Lighttpd plugins output chains of chuks of a few types, which include strings, byte arrays and file fragments. The `tar` format is well-fit for this, since it can be built on-the fly: it's got file headers followed by data [1]. This even allows for `sendfile` to be used by the server.

The NGINX API seemed similar to this, but somewhat more involved.

[1] Things are a bit more fiddly than this because of the need to handle long file names and symlinks, but it's still streamable.

**Comment by aperezdc** [comment]
Yes, the Nginx API might be a bit more complex, but things are still manageable and streaming the created tarball to the client in chunks is certainly possible. My concern is about the complexity... anyhoo, I will revisit this idea after tagging the `0.4.0` release.

**Comment by mapsic** [comment]
I can't find fancyindex.conf in my centos6.6

**Comment by lnicola** [comment]
@mapsic `ngx-fancyindex` is an `Nginx` plugin. Until recently you had to build these plugins together with `Nginx` in order to use them, although it now supports shared linking. I don't know how it's done now.

**Comment by aperezdc** [comment]
@lnicola, @mapsic: Presently it's possible to build modules for Nginx and load them dynamically. This feature was [introduced in Nginx 1.9.11](https://www.nginx.com/resources/wiki/extending/converting/), and is supported by `ngx-fancyindex`.

**Comment by annulen** [comment]
Any updates on this topic? Seems like nobody has implemented it yet.

Though I definitely sympathize to your concerns about complexity and possible huge data transfers (which can result in many kinds of nasty things if it's enabled carelessly). I think it would be fine with me to offload it into backend script. 

However, it would be great if user could somehow add "download directory" link into the HTML template of file listing using config option. Something like
```
fancyindex_download_dir on;
fancyindex_download_dir_request http://localhost:3333/myscript?dir=
```
What do you think?
